qiskit_code_assistant_jupyterlab/handlers.py

```diff
@@ -33,35 +33,177 @@
     "service_url": "http://localhost",
     "api_token": "",
     "is_openai": False,
+    "selected_credential": None,
+    "using_env_var": False,  # Track if env var is overriding selection
 }
 
 
 def update_token(token):
     if token:
         runtime_configs["api_token"] = token
+        # When manually setting token, update selected credential to match
+        runtime_configs["selected_credential"] = "qiskit-code-assistant"
+        # Save to both qiskit-ibm.json and preferences
         QiskitRuntimeService.save_account(
             channel="ibm_quantum",
             name="qiskit-code-assistant",
             token=token,
             overwrite=True,
         )
+        save_selected_credential("qiskit-code-assistant")
+        print("Manually entered token saved as 'qiskit-code-assistant' credential")
```

Comment:
i keep getting this error:

<img width="639" height="80" alt="image" src="https://github.com/user-attachments/assets/556b2c42-059f-4dd2-a661-f3caae21012c" />